analysis_options.yaml

```diff
@@ -7,7 +7,26 @@ linter:
   rules:
     # Util classes
     avoid_classes_with_only_static_members: false
-
     non_constant_identifier_names: false
-
     constant_identifier_names: false
+    always_declare_return_types: true
```

Comment:
rules are by default enabled unless explicitly disabled.

Reply:
Nope. You can see the changes. Earlier when these rules are not mentioned as true there were no issue otherwise CI would fail at the `flutter analyze .` step.

Reply:
The default rules and requirements are already enabled. Explicitly enabling more rules would cause more errors and stricter rules for building.

Reply:
Linters are basically meant for that only to enforce best practices. And I am not explicitly enabling rules from my side, the above was mentioned in the issue.

Reply:
> Nope. You can see the changes. Earlier when these rules are not mentioned as true there were no issue otherwise CI would fail at the `flutter analyze .` step.

So, you are saying that if any rule is not set to be true. By default they are not checked by linter ?

Reply:
So this confusion everyone had, you are right that we have to enable them as the specs of analyzer is designed in this way. 

But actually, I meant that you don't need to write enable statements since they are already included by `flutter_lints` package (check include statement in `analysis_options.yaml:1`) and in turn `flutter_lints` uses `dart-lang/lints`'s recommended linter rules.

I hope that clears the doubt ?

Reply:
@manjotsidhu 

> So this confusion everyone had, you are right that we have to enable them as the specs of analyzer is designed in this way.
> 
> But actually, I meant that you don't need to **write** enable statements since they are already included by `flutter_lints` package (check include statement in `analysis_options.yaml:1`) and in turn `flutter_lints` uses `dart-lang/lints`'s recommended linter rules.
> 
> I hope that clears the doubt ?

Yeah, got it @manjotsidhu. So, I am closing this PR.

Reply:
> > Nope. You can see the changes. Earlier when these rules are not mentioned as true there were no issue otherwise CI would fail at the `flutter analyze .` step.
> 
> So, you are saying that if any rule is not set to be true. By default they are not checked by linter ?

Yes, all the linter rules specified in the issue are not checked by default. We have to specify them.